src/extension/background-script/actions/nostr/helpers.ts

```diff
@@ -12,29 +13,37 @@ export async function hasPermissionFor(method: string, host: string) {
   const allowance = await db.allowances.get({
     host,
   });
+  const accountId = state.getState().currentAccountId;
 
-  if (!allowance?.id) {
+  if (!allowance?.id || !accountId) {
     return false;
   }
 
-  const findPermission = await db.permissions.get({
+  let findPermission = await db.permissions.get({
```

Comment:
I think we could directly query here the permission for that host, method, and and account.

Reply:
But then we would be leaving the old permissions with no accountId

Reply:
(Saw your other comment 👍)